compiler/noirc_evaluator/src/ssa/opt/inlining.rs

```diff
@@ -68,8 +68,16 @@ impl Ssa {
         // instead of inlining the "leaf" functions, moving up towards the entry point.
         self.functions = btree_map(inline_targets, |entry_point| {
             let function = &self.functions[&entry_point];
-            let new_function =
-                function.inlined(&self, inline_no_predicates_functions, inline_infos);
+            let should_inline_call =
+                |_context: &PerFunctionContext, ssa: &Ssa, called_func_id: FunctionId| -> bool {
```

Comment:
Can you remove the `_context` arg if this is forcing us to expose `PerFunctionContext` beyond the inlining pass?